### DISCUSSION TITLE
Pass mode to vitest vsc plugin

### DISCUSSION THREAD

**Discussion by waldemarennsaed** [vitest-dev/vscode#397] [Q&A] [answered]
Is it possible to configure the vitest plugin in a way to pass a vite `mode`?

Currently I am executing my unit-tests by running this command (a script in my `package.json`):

```vitest run --mode=test-unit```

I need the `mode` `test-unit` to be passed to the VSC plugin, otherwise some tests are failing.

Is that possible?

**Comment by sheremet-va** [comment]
Do you have a config file? By default, the extension uses that. If there are no configs, it uses a script from `scripts`  that starts with `vitest`

Currently, there is no way to pass down arguments manually, but you can use `process.env.VITEST_VSCODE` in the config file

**Reply by waldemarennsaed** [reply]
Well, in my `vite.config.mts` file I have the following `test` block defined:

```
test: {
      coverage: {
        enabled: false,
        exclude: ['**/*sampleData.ts', '**/*mock*.ts'],
        include: ['src/**/*.{ts,vue}'],
        provider: 'v8',
        reporter: ['html', 'text']
      },
      css: true,
      environment: 'jsdom',
      globals: true,
      onConsoleLog (log) {
        if (loggedMessagesToIgnore.some((msg) => log.includes(msg))) {
          return false
        }
      },
      server: {
        deps: {
          inline: ['vuetify', 'moment']
        }
      },
      setupFiles: ['./public/config.js', './public/secrets.js', './tests/setup.ts']
    }
```

And my `package.json` scripts for unit-testing with vitest are defined as following:

```
{
    "test:unit": "vitest run --mode=test-unit --coverage.enabled --coverage.reporter=text-summary",
    "test:unit:coverage": "vitest run --mode=test-unit --coverage.enabled --coverage.reporter=text-summary --coverage.reporter=cobertura",
    "test:unit:ui": "vitest --mode=test-unit --ui",
    "test:watch": "vitest --mode=test-unit",
}
```

When executing `npm run test:unit` -> `vitest run --mode=test-unit --coverage ...` it works just fine. But hitting the double-play-button on the extension, makes one test suite fail:

![Bildschirmfoto 2024-05-23 um 11 48 02](https://github.com/vitest-dev/vscode/assets/94853611/d95e4446-8742-4e4c-bbe7-88550f2d1f40)

**Reply by sheremet-va** [reply]
You can specify `mode` in `test` object in your config file then

**Reply by waldemarennsaed** [reply]
I get a TS error and also my tests still fail when using the double-play-button:

![Bildschirmfoto 2024-05-23 um 12 02 22](https://github.com/vitest-dev/vscode/assets/94853611/2fdf6d9a-027a-498c-9f3e-167d81eee119)

**Reply by sheremet-va** [reply]
Sorry, the `mode` property is top level:

```ts
export default defineConfig({
  mode: 'test-unit',
})
```

If you share the config with Vite, you can set it conditionally: `process.env.VITEST ? 'test-unit' : 'staging'`

**Reply by waldemarennsaed** [reply]
Just tried this out, seems not to work for me.

**Comment by giacomocerquone** [comment, marked as answer]
For people coming to see this, I had the same problem and you can pass cli flags through the extension's settings

<img width="1016" height="166" alt="image" src="https://github.com/user-attachments/assets/383f799b-0bd8-4853-a964-5a8fd12bc50b" />
